engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -57,16 +66,20 @@ private class Factory(
 ) {
     private val definitionNamesToService: Map<String, Service> = makeDefinitionNamesToService()
     private val coordinatesToService: Map<FieldCoordinates, Service> = makeCoordinatesToService()
+    private val typeRenameInstructions = makeTypeRenameInstructions()
+    private val typesToServices: Map<String, Set<Service>> =
+        TypeOwnership(overallSchema, services, typeRenameInstructions)
```

Comment:
Ok so what we're doing here is actually figuring out which services define a given type.

Before this change, we never considered an overall type that was shared across multiple services i.e. `definitionNamesToService: Map<String, Service>`

But now we have `typesToService: Map<String, Set<Service>>`, please fix the naming for me, or do something.

The issue with the `Map<String, Service>` was that in the shared service we have, we never define underlying types. So in this factory, we could never find the underlying type if it was shared. So here we store all possible services and look at _one_ given underlying type.

---

Somewhere else, we should be validating that ALL underlying schemas are valid WRT to their overall type i.e. if the SHARED overall type is

```graphql
type Test {
  id: ID
 name: String
}
```

We don't have two underlying definitions that are:

```graphql
# in service A
type Test {
  id: ID
  name: String
}
# in service B
type Test {
  id: ID
}
```